### PR TITLE
fix: fix bug when trying to re-try EOA connections from auth card

### PR DESCRIPTION
**Pull request by iykazrji**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

# Testing plan
- Simulate an EOA connection failure (WalletConnect, Metamask, CoinbaseWallet)
- Hit on `Try Again` from the Auth Card
- This should re-start the attempted EOA connection process.

---

## PR-Codex overview
This PR focuses on enhancing the authentication process by introducing a new hook `useConnectEOA`, improving error handling for Wallet Connect, and optimizing the connection flow for EOA and Wallet Connect in the authentication cards.

### Detailed summary
- Added `useConnectEOA` hook for EOA connection management.
- Introduced `clearWalletConnectStore` function to handle Wallet Connect state.
- Updated `ConnectionError` component to disconnect on error.
- Refactored `AuthCardContent` to use `disconnect` instead of `signer.disconnect`.
- Enhanced `EoaConnectCard` and `WalletConnectCard` to retry connections.
- Removed redundant connection logic in `EoaPickCard`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/react/src/components/auth/card/error/connection-error.tsx [hunk 1]
```diff
@@ -86,3 +103,23 @@ export const ConnectionError = ({
...
+    const store = txn.objectStore("keyvaluestorage");
+
+    // Clear Store
+    store.clear();
+  };
+
+  dbOpenRequest.onerror = () => {
+    console.error("Error opening Wallet Connect DB. Cannot clear store.");
+  };
+};
```
Comment (iykazrji) [inline account-kit/react/src/components/auth/card/error/connection-error.tsx:125]: Adding this function to clear the IndexedDB store wallet connect  populates on attempted authentication. This is to address the fact that we could get into continuous error loops when an auth attempt fails.

**Review by rthomare — DISMISSED** [review]

File: account-kit/react/src/components/auth/card/error/connection-error.tsx (hunk 1 above)
Comment (moldy530) [reply account-kit/react/src/components/auth/card/error/connection-error.tsx:125]: oh added a comment on the other PR but we might want to add this to the `disconnect` action from `account-kit/core`

Comment (moldy530) [reply account-kit/react/src/components/auth/card/error/connection-error.tsx:125]: actually... looking at the definition for wallet connect providers, https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/universal-provider/src/UniversalProvider.ts#L546

the session should not be getting used on retry if we're using the `disconnect` action from core which calls the `wagmi_disconnect` method. That method then calls the connectors disconnect which eventually clears the wallet connect session.

Comment (iykazrji) [reply account-kit/react/src/components/auth/card/error/connection-error.tsx:125]: From my tests, this doesn't seem to be the case. I'd be sending a video recording to explain the behaviour calling `disconnect` from `@account-kit/core`

**Pull request closed without merging**